BoxContentSDK/BoxContentSDK/BOXURLRequestSerialization.m

```diff
@@ -1166,6 +1179,10 @@ - (NSInteger)read:(uint8_t *)buffer
         NSInteger numberOfBytesRead = 0;
 
         numberOfBytesRead = [self.inputStream read:&buffer[totalNumberOfBytesRead] maxLength:(length - (NSUInteger)totalNumberOfBytesRead)];
+        if(self.contentProcessor) {
+            NSData *bodyData = [NSData dataWithBytes:(const void *)(&buffer[0] + totalNumberOfBytesRead) length:numberOfBytesRead];
```

Comment:
Could use `buffer` rather than `&buffer[0]`

Reply:
If we know that `processData:` completes synchronously, we should be able to use the non-copying NSData constructor here, and avoid a little work.

Reply:
I can pass raw pointers without converting to NSData to save on the copy but I thought passing around the chunk as NSData would be a better interface for the contentProcessor

Reply:
I was thinking about `+[NSData dataWithBytesNoCopy:length:]`